### ISSUE TITLE
Report from Student - LTI Authentification Does Not Persist

### ISSUE THREAD

**Issue by kwahlenmayer** [Simon-Initiative/oli-torus#4214]
A great addition to TORUS would be a longer time out window when reviewing materials. I find myself timing out when reading and by the time I get to the questions I have to log back in.

----------------------------------------------
Timestamp: 2023-09-15 03:42:06.909208+00:00 UTC UTC
Ip Address: 0.0.0.0.0.65535.32512.1
Location: https://proton.oli.cmu.edu/sections/chem04b72042_fall_2023_0hbtx/page/01e42_conjugate_acids_and_bases_8chhv

WEB BROWSER
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36
Accept: */*
Language: en-US,en;q=0.9

CAPABILITIES
Cookies Enabled: true

USER ACCOUNT
Name:
Email:
Created:

**Comment by kwahlenmayer** [comment]
Hi Gabriel,

Thanks for emailing about this. Your sign-in to OLI Torus is through your LMS / school's system so should be persisting as long as you can access your school's system. If that does not seem to be happening, let me know more details. I will pass this along to our larger team for review.

Let me know if you have any questions!
 
Kim
OLI Support

**Comment by kwahlenmayer** [comment]
This type of report does happen every once in a while where a student reports they are being signed out repeatedly. 

It is to be determined whether timing out of sign-in impacts quiz submission, blank answers, double attempt #s (i.e. two attempt #3s).

**Comment by kwahlenmayer** [comment]
UPDATE: Student reports that the Canvas sign-in persists, but Torus does not. This is resolved by switching from Edge to Chrome.

**Comment by kwahlenmayer** [comment]
UPDATE: I tested the persistence of authentication from Canvas to Torus. I submitted a quiz at 11 AM successfully. I started a new quiz. I went back at 3 PM, entered answers, submitted - the submitting button grayed and remained grayed but never submitted. When I clicked on course content, proton.oli.cmu.edu appeared. I went back to Canvas, clicked through to OLI Torus again, navigated to the quiz, saw that my quiz did not submit.

The Canvas sign-in persisted, the OLI Torus sign-in expired within 3 hours. The quiz had no message of not being signed in or error upon submit (leading student to potentially believe quiz was submitted). The refresh of OLI Torus dropped student into the sign-in page with no message about needing to sign back into school's system to access course.

Ideally:
the OLI Torus sign-in would persist longer than 3 hours or serve up a "sign back in to your school's system" upon interaction by user
